### PR TITLE
Sprint 1

### PR DESCRIPTION
**Pull request by AlauddinSaikat71**

### CODE REVIEW

File: controllers/users.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	Password string `json:"password" binding:"required"`
+}
+
+type UpdateUserInput struct {
+	Name     string `json:"name" `
+	Email    string `json:"email" `
+	Phone    string `json:"phone" `
+	Password string `json:"password" `
+}
+
```
Comment (rafiulgits) [inline controllers/users.go]: move these struct inside models folder, create a subfolder `dtos` inside models and move there. Make controllers fresh

File: models/setup.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+var DB *gorm.DB
+
+func ConnectDatabase() {
+	database, err := gorm.Open("sqlite3", "test.db")
+
+	if err != nil {
+		panic("Failed to connect to database")
+	}
+	database.AutoMigrate(&User{})
+	DB = database
```
Comment (rafiulgits) [inline models/setup.go]: this file should be inside `db` package. A package or folder named `db` in root folder

**Review by rafiulgits — CHANGES_REQUESTED** [review]
Please update those things

File: controllers/users.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	// Validate input
+	var input dtos.UpdateUserInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	// Get model if exist
+	var user models.User
+	if err := db.DB.Where("id = ?", c.Param("id")).First(&user).Error; err != nil {
```
Comment (rafiulgits) [inline controllers/users.go]: you are accessing `db.DB.Where()`, how your system or code recognize that you are searching in User table. Suppose we have multiple table exist including users table, then how your query recognize on which table it should query?

**Review by rafiulgits — COMMENTED** [review]
Can you please describe my question on database?

File: controllers/users.go (hunk 1 above)
Comment (AlauddinSaikat71) [reply controllers/users.go]: change done

File: models/setup.go (hunk 2 above)
Comment (AlauddinSaikat71) [reply models/setup.go]: change done

File: controllers/users.go (hunk 3 above)
Comment (AlauddinSaikat71) [reply controllers/users.go]: this query is taking the first "user" model .
.First(&user)

and user is a variable of User type

Comment (rafiulgits) [reply controllers/users.go]: Oh. Okey. But this is little bit confusing. We can simply create some instances like `userDbContext` or separate all database operation into a repository layer.

Comment (AlauddinSaikat71) [reply controllers/users.go]: ### userDbContext ###
will contain the user table ??

Comment (rafiulgits) [reply controllers/users.go]: yes. like `userDbContext := db.Table(domains.UserTable)`. Is it more readable for anyone who will letter handle this project or want to contribute in this project?

Comment (AlauddinSaikat71) [reply controllers/users.go]: what if store all users like GET operation and then run query on the list?

Comment (rafiulgits) [reply controllers/users.go]: don't get you

Comment (AlauddinSaikat71) [reply controllers/users.go]:
```
func FindUsers(c *gin.Context) {
	var users []models.User
	db.DB.Find(&users)
}
```
 i want to run query in
```users ```

Comment (rafiulgits) [reply controllers/users.go]: I already agreed with your concern. I am just talking about code readability. When someone read this code, it is very much possible to get confused.

Comment (rafiulgits) [reply controllers/users.go]: Should I do merge?

Comment (AlauddinSaikat71) [reply controllers/users.go]: i am confused about 
```
domains.UserTable
```

Comment (rafiulgits) [reply controllers/users.go]: oh, this is just a string, where UserTable = "users"

Comment (rafiulgits) [reply controllers/users.go]: name of that table, defined in a constant

**Pull request merged**